modules/gdscript/gdscript_parser.cpp

```diff
@@ -4257,20 +4284,35 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 		variable->export_info.type = enum_type;
 
 		if (!export_type.is_variant() && (export_type.kind != DataType::BUILTIN || export_type.builtin_type != enum_type)) {
-			push_error(vformat(R"("@export_enum" annotation requires a variable of type "int" or "String" but type "%s" was given instead.)", export_type.to_string()), variable);
+			Vector<Variant::Type> expected_types;
+			expected_types.push_back(Variant::INT);
+			expected_types.push_back(Variant::STRING);
```

Comment:
```suggestion
			Vector<Variant::Type> expected_types = { Variant::INT, Variant::STRING };
```

You can use the syntax, if I remember correctly. The same in other places.